### PR TITLE
Remove expect.js, media-typer, underscore

### PR DESCRIPTION
**Pull request by pluma**
Also move dev dependencies into devDependencies. Closes #15657.

### Scope & Purpose

This PR removes unused Foxx dependencies that were undocumented prior to the deprecation of bundled Foxx dependencies.

Additionally ansi_up, eslint and `@xmldom/xmldom` have been moved into devDependencies to allow filtering them out in the build/release process as these dependencies are only relevant during development.

- [x] :hammer: Refactoring/simplification

### Checklist

- [x] :book: CHANGELOG entry made

#### Related Information

- [x] Enterprise PR: https://github.com/arangodb/enterprise/pull/975
- [x] Jira ticket: APM-277

### CODE REVIEW

**Comment by dothebart** [comment]
are underscore and lodash the same?

**Comment by pluma** [comment]
Underscore and lodash are mostly equivalent, yes. We migrated our own code off underscore in the past. I think some early 2.x documentation may have used underscore as well, but only lodash was ever documented for 3.x and (I think) 2.8.

**Review by mpoeter — APPROVED** [review]

**Review by Zyqsempai — APPROVED** [review]

**Review by adityamukho — APPROVED** [review]
LGTM

File: CHANGELOG [hunk 1]
```diff
@@ -1,7 +1,10 @@
 devel
 -----
 
-* APM-84: Added option to spill intermediate AQL query results from RAM to
+* Removed internal JavaScript dependencies "expect.js", "media-typer" and
+  "underscore". We recommend always bundling your own copy of third-party
+  modules as all previously included third-party modules are now considered
+  deprecated and may be removed in future versions of ArangoD* APM-84: Added option to spill intermediate AQL query results from RAM to
```
Comment (hkernbach) [inline CHANGELOG:7]:
```suggestion
  deprecated and may be removed in future versions of ArangoDB
  
  * APM-84: Added option to spill intermediate AQL query results from RAM to
```

Comment (hkernbach) [reply CHANGELOG:7]: Format issue.

**Review by hkernbach — APPROVED** [review]
LGTM. Only one minor malformat issue found in the changelog.

**Pull request merged**